fix(common): incorrect error type for XHR errors in `TestRequest` (#36082)
angular/angular: aio/content/guide/deprecations.md

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update. 

{@a testrequest-errorevent}
### `TestRequest` accepting `ErrorEvent`

Angular provides utilities for testing `HttpClient`. The `TestRequest` class from
`@angular/common/http/testing` mocks HTTP request objects for use with `HttpTestingController`.

`TestRequest` provides an API for simulating an HTTP response with an error. In earlier versions
of Angular, this API accepted objects of type `ErrorEvent`, which does not match the type of

> error event being returned by browsers natively. If you use `ErrorEvent` with `TestRequest`,

Reviewer: ```suggestion
error event that browsers return natively. If you use `ErrorEvent` with `TestRequest`,
```
Passive voice. Hard to avoid sometimes!

Author: thanks. that's much better 👍

> you should switch to `ProgressEvent`. A future version of Angular will remove support for
> using `ErrorEvent` with `TestRequest`.

Reviewer: ```suggestion
you should switch to `ProgressEvent`.
```

In general, I advise against statements such as "a future version..." or "in the future..." They often don't add as much value as we think--often, readers care about the state of the product as it is right now, not how it will be at some point in the future.

In this case, I think there's a second reason to remove this sentence: We're already in the deprecations topic. The fact that something might be removed is probably well understood. :)

> Here is an example using a `ProgressEvent`:

Reviewer 2: ```suggestion
The following example demonstrates use of `ProgressEvent` with `HttpTestingController`:
```

- Clone the resulting value so that it has a new identity.
- Explicitly call [`ChangeDetectorRef.detectChanges()`](api/core/ChangeDetectorRef#detectchanges) to force the update. 

{@a testrequest-errorevent}
### `TestRequest` accepting `ErrorEvent`

> The `@angular/common` package provides utilities for testing the `HttpClient`. One part of these
> testing utilities is the `TestRequest` class. It serves as mock for HTTP requests and is primarily
> used in the `HttpTestingController`.

Reviewer 2: ```suggestion
The `@angular/common` package provides utilities for testing `HttpClient`. The `TestRequest` class from `@angular/common/http/testing` mocks HTTP request objects for use with `HttpTestingController.
```

Author: Yeah, that's definitely better. thx!

> `TestRequest` provides an interface for resolving to mocked content, or simulating an errored http
> request. Currently, the method for simulating an error accepts objects of type `ErrorEvent`. This
> does not match with error events that are returned by browsers natively, so we intend to no longer
> accept `ErrorEvent`.

Reviewer 2: ```suggestion
`TestRequest` provides an API for simulating an HTTP response with an error. In earlier versions of Angular, this API accepted objects of type `ErrorEvent`, which
does not match the type of error event that are returned by browsers natively. If you use `ErrorEvent` with `TestRequest`, you should switch to `ProgressEvent`. A future version of Angular will remove support for using `ErrorEvent` with `TestRequest`. 
```

Author: Thanks! way better 👍
